hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceImpl.java

```diff
@@ -132,9 +132,10 @@ protected HazelcastInstanceImpl(String name, Config config, NodeContext nodeCont
 
         try {
             logger = node.getLogger(getClass().getName());
-            lifecycleService.fireLifecycleEvent(STARTING);
```

Comment:
Imagine you have a `HazelcastInstanceAware` listener. 

I don't see much usage for the `STARTING`. It says something: "We have initiated `HazelcastInstance` start, but it's not quite ready yet".